Automattic/wp-calypso
# Domains: Show the transfer away eligibility date in the UI

#### Changes proposed in this Pull Request

- Add the date to the assembler
- Fix tests
- Add new UI element when transfer lock is present

#### Testing instructions

- Update contact info for a SRS/HRS domain in calypso w/o opting out of the transfer lock
- Try to transfer to a different registrar
- Should see the transfer lock UI with transfer eligibility date
- Now update the same contact info but opt out of the lock
- Try to transfer to a different registrar
- Shouldn't see the transfer lock UI with transfer eligibility date

Depends on D20715-code.
